### PR TITLE
Adds test for `cub::PtxVersion`

### PR DESCRIPTION
**Pull request by elstehle**
## Description

Adds test for `cub::PtxVersion`, which is a core component of CUB's dispatch mechanism.

## Checklist

- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cub/test/catch2_test_util_device.cu [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+CUB_RUNTIME_FUNCTION static cudaError_t
+GetCudaArchFromKernel(void* d_temp_storage, size_t& temp_storage_bytes, int* d_kernel_cuda_arch, int* ptx_version, cudaStream_t stream = 0)
+{
+  if (d_temp_storage == nullptr)
+  {
+    temp_storage_bytes = 1;
+    return cudaSuccess;
+  }
```
Comment (miscco) [inline cub/test/catch2_test_util_device.cu:52]: Do we really need those arguments or is the kernel launch ignored otherwise?

Comment (elstehle) [reply cub/test/catch2_test_util_device.cu:52]: I think the CDP/graph launch helper expects the default CUB behaviour of ptr-to-sctrach-space, sctrach-space-reference

File: cub/test/catch2_test_util_device.cu [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  // Host cub::PtxVersion
+  int host_ptx_version{};
+  cub::PtxVersion(host_ptx_version);
+
+  REQUIRE(*ptx_version == kernel_cuda_arch);
+  REQUIRE(host_ptx_version == kernel_cuda_arch);
+
+  std::cout << "ptx_version: " << *ptx_version << "\n";
+  std::cout << "kernel_cuda_arch: " << kernel_cuda_arch << "\n";
```
Comment (gevtushenko) [inline cub/test/catch2_test_util_device.cu:82]: suggestion: Ideally, tests should not print anything except errors. If the equality doesn't hold, Catch2 will print both values anyways. 
```suggestion
```

File: cub/test/catch2_test_util_device.cu [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  // Query the arch the kernel was actually compiled for
+  get_cuda_arch_from_kernel(thrust::raw_pointer_cast(cuda_arch.data()), ptx_version);
+  int kernel_cuda_arch = cuda_arch[0];
+
+  // Host cub::PtxVersion
+  int host_ptx_version{};
+  cub::PtxVersion(host_ptx_version);
+
+  REQUIRE(*ptx_version == kernel_cuda_arch);
+  REQUIRE(host_ptx_version == kernel_cuda_arch);
```
Comment (gevtushenko) [inline cub/test/catch2_test_util_device.cu:79]: suggestion: just to make sure, can you compare that both are not zero? Currently, if `cub::PtxVersion` just exists and kernel is not launched, test will pass.

File: cub/test/catch2_test_util_device.cu [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+#include <thrust/detail/raw_pointer_cast.h>
+#include <thrust/device_vector.h>
+
+#include "catch2_test_helper.h"
+#include "catch2_test_launch_helper.h"
+
+CUB_NAMESPACE_BEGIN
+
+CUB_DETAIL_KERNEL_ATTRIBUTES void WritePtxVersionKernel(int* d_kernel_cuda_arch)
```
Comment (gevtushenko) [inline cub/test/catch2_test_util_device.cu:40]: suggestion: we are trying to get away from camel case.

**Review by gevtushenko — APPROVED** [review]

File: cub/test/catch2_test_util_device.cu (hunk 3 above)
Comment (elstehle) [reply cub/test/catch2_test_util_device.cu:83]: Great catch! Added a check to verify they are not `0`.

**Pull request merged**